### ISSUE TITLE
Batch Process - Sql Connection Failures

### ISSUE THREAD

**Issue by joetherod** [aspnet/Mvc#8492]
I ported over a .NET Windows Service to MVC Core using HostedService. It is a batch process that runs daily and does a lot of reads against SQL and MongoDB. SQL is the painstaken issue. Declared service that does the sql calls as singletons, container scoped, thread local with no avail. The process also spins off about 500 threads. I had no issues with in .NET 4.5 running on windows. Now, it consumes all the memory and high CPU and I cant around the SQL connection issues. Its able to connect many times and then it just dies on 2 issues,

"BeginExecuteReader requires an open and available Connection. The connection's current state: Broken."
or
Unable to connect to sql server. (Connection pool exhausted)

The sql calls are Dapper based, wrapper around ADO.NET.
Core seems to issues with managing sql connections and memory leaks.
Tried so many combinations of registrations and limiting the number of uses of the service that has the sql connection. No luck.

I am using the latest .NET Core 2.1.

**Comment by nh43de** [comment]
Can you provide some sample code or minimal reproduction?

**Comment by joetherod** [comment]
I was able to fix the sql issue and exhausting connections. The registration of services was not being honored since I didnt explicitly define a scope off the service provider. As per example when using a hosted service.
Eg.
using (var scope = Services.CreateScope())
        {
            var scopedProcessingService = 
                scope.ServiceProvider
                    .GetRequiredService<IScopedProcessingService>();

            scopedProcessingService.DoWork();
        }

My issue now is with memory consumption. In .NET you are able to configure memory limits.
<system.runtime.caching>
        <memoryCache>
            <namedCaches>
                <add name="Default" cacheMemoryLimitMegabytes="0" physicalMemoryLimitPercentage="50" />
            </namedCaches>
        </memoryCache>
    </system.runtime.caching>

In core you dont have app memory limit configurations available as far as what I can find with docker containers hosted in linux.
For memory cache, the limitations are on the entry of items, not the total availability from what I read of in memory cache options, Size, SizeLimit. Those are on per entry.

I also tried adding this to the project file:
<PropertyGroup>
    <ServerGarbageCollection>false</ServerGarbageCollection>
  </PropertyGroup>

My app is a batch process, the reads a lot of sql data, mongodb cursor, Elastic Searches, and use of in memory cache.

Issue is with in memory cache limitations, cleaning up if using more than available and garbage collection. 

Any ideas on options? This all worked fine in a .NET windows service.
